src/reaction.jl

```diff
@@ -343,6 +343,42 @@ end
 MT.is_diff_equation(rx::Reaction) = false
 MT.is_alg_equation(rx::Reaction) = false
 
+"""
+    get_symbolics(set, rx::Reaction)
+
+Returns all symbolic variables that are part of a reaction. This includes all variables 
+encountered in:
+    - Rates.
+    - Among substrates and products.
+    - Among stoichiometries.
+    - Among potential noise scaling metadata.
+"""
+function get_symbolics(rx::Reaction)
+    return ModelingToolkit.get_variables!([], rx)
+end
+
+"""
+    get_variables!(set, rx::Reaction)
+
+Adds all symbolic variables that are part of a reaction to set. This includes all variables 
+encountered in:
+    - Rates.
+    - Among substrates and products.
+    - Among stoichiometries.
+    - Among potential noise scaling metadata.
+"""
+function ModelingToolkit.get_variables!(set, rx::Reaction)
+    get_variables!(set, rx.rate)
+    append!(set, rx.substrates)
+    append!(set, rx.products)
```

Comment:
```suggestion
    foreach(sub -> push!(set, sub), rx.substrates)
    foreach(prod -> push!(set, prod), rx.products)
```
`append!` would restrict this function to only work on vectors (i.e. `Sets` wouldn't work).

Reply:
That make sense, good catch.